chore: Remove default gas settings

spalladino opened the pull request:
Deletes default fee in gas settings, and makes `maxFeePerGas` required for initializing a `GasSettings` instance.

Built on top of https://github.com/AztecProtocol/aztec-packages/pull/10105.

spalladino committed:
chore: Remove default gas settings

spalladino commented:
@LHerskind this PR _builds_, but I'm waiting until your PR is green to start fighting the CI on this one. Feel free to start reviewing in the meantime if you want to.

LHerskind commented:
> @LHerskind this PR _builds_, but I'm waiting until your PR is green to start fighting the CI on this one. Feel free to start reviewing in the meantime if you want to.

Understandable. The other is fighting ci on its own at the moment. Will take a look in the morning 🫡

spalladino commented:
> Will take a look in the morning

No rush at all. Thanks!

LHerskind reviewed (commented):
Took a look. Will probably look again in the morning.

spalladino commented on yarn-project/aztec.js/src/contract/base_contract_interaction.ts:
```diff
@@ -89,29 +89,47 @@ export abstract class BaseContractInteraction {
+      const feeForEstimation: FeeOptions = { paymentMethod, gasSettings };
+      const txRequest = await this.wallet.createTxExecutionRequest({ ...request, fee: feeForEstimation });
       const simulationResult = await this.wallet.simulateTx(txRequest, true);
       const { totalGas: gasLimits, teardownGas: teardownGasLimits } = getGasLimits(simulationResult);
```
Sounds good. I took your `estimateGasPad` and moved it inside the new `UserFeeOpts`.

spalladino commented on yarn-project/aztec.js/src/contract/deploy_method.ts:
```diff
@@ -133,7 +130,9 @@ export class DeployMethod<TContract extends ContractBase = Contract> extends Bas
-  protected async getDeploymentFunctionCalls(options: DeployOptions = {}): Promise<ExecutionRequestInit> {
+  protected async getDeploymentFunctionCalls(
+    options: DeployOptions = {},
+  ): Promise<Pick<ExecutionRequestInit, 'calls' | 'authWitnesses' | 'packedArguments'>> {
```
Yep, where this is called from, we accumulate all calls along with their authwits and packed args.

spalladino commented on yarn-project/end-to-end/src/e2e_fees/fees_test.ts:
```diff
@@ -135,6 +134,7 @@ export class FeesTest {
       async ({ accountKeys }, { pxe, aztecNode, aztecNodeConfig }) => {
         this.pxe = pxe;
         this.aztecNode = aztecNode;
+        this.gasSettings = GasSettings.default({ maxFeesPerGas: await this.aztecNode.getCurrentBaseFees() });
```
I assumed that these tests sent few enough txs that prices would not spike. Would it be safer to set these fees to twice the current base fees, just in case?

spalladino committed:
Pad max gas fees in fees test

spalladino commented:
@LHerskind addressed all comments but the ones on the hardcoded `(10,10)` fees. It's difficult to properly fix them without a big overhaul of our tests. Do you think we can set a high enough constant to get us past this?

spalladino committed:
Fix aztecjs test

spalladino committed:
Bump FEE_JUICE_INITIAL_MINT

spalladino committed:
Bump FEE_FUNDING_FOR_TESTER_ACCOUNT and move out of global constants

LHerskind reviewed (approved):
🚢
